service/src/main/kotlin/fi/espoo/evaka/application/ApplicationControllerCitizen.kt

```diff
@@ -822,7 +822,20 @@ class ApplicationControllerCitizen(
                 )
                 .filter { it.value.isPermitted() }
                 .keys
-        return canAccept.intersect(canReject)
+
+        return tx.createQuery {
+                sql(
+                    """
+            SELECT a.id
+            FROM application a
+            WHERE a.id = ANY(${bind(canAccept.intersect(canReject))}) AND EXISTS(
+                SELECT FROM decision d
+                WHERE d.application_id = a.id AND d.status = 'PENDING'
```

Comment:
Omaan silmään tämä olis siistimpi:
```suggestion
            SELECT a.id
            FROM application a
            JOIN decision d ON d.application_id = a.id
            WHERE a.id = ANY(${bind(canAccept.intersect(canReject))}) AND d.status = 'PENDING'
```

Reply:
Joo, on siin tosiaan toSet niin duplikaatitkaan ei haittaa